db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBClient.java

```diff
@@ -59,13 +63,29 @@ public static EventStoreDBClient create(EventStoreDBClientSettings settings) {
     }
 
     public CompletableFuture<ReadResult> readStream(String streamName, long maxCount, ReadStreamOptions options) {
+        return readEventsFromPublisher(this.streamEvents(streamName, maxCount, options));
+    }
+
+    public Publisher<ResolvedEvent> streamEvents(String streamName) {
```

Comment:
**SUGGESTION:** Thinking out loud, `streamEvents` may be confused with the general stream concept. Maybe we could name those methods like `readStreamAsync`.

@dpasek-senacor, @YoEight, @hayley-jean, @jageall thoughts?

Reply:
`readStreamAsync()` would be fine for me.

Reply:
Being precise... the other `readStream()` methods are also async (using CompletableFuture) 😉 

Internally we discussed `readStreamReactive` or `readStreamReactively` but it felt pretty clunky, `readStreamAsync` is definetely better.

Reply:
Yup, I know, it's tricky, for `subscribeToStream` would also work well, but it's also clashing with the subscription feature... 😉

 `readStreamReactive` is also okay for me. We'll need to discuss how to make this API more self-explanatory and consistent in the long term. But that's another story. 😉

Reply:
`readStreamReactive` seems sensible to me

Reply:
@YoEight @hayley-jean @jageall 
Any more thoughts on the naming of the reactive methods?
Current two contenders: 
- `readStreamAsync` / `readAllAsync` 
- `readStreamReactive` / `readAllReactive`

Reply:
we are choosing the `Reactive` suffix!

Reply:
Done!